scene/gui/tree.cpp

```diff
@@ -3455,30 +3459,48 @@ int Tree::get_item_offset(TreeItem *p_item) const {
 }
 
 void Tree::ensure_cursor_is_visible() {
-
-	if (!is_inside_tree())
+	if (!is_inside_tree()) {
 		return;
+	}
+	if (!selected_item || (selected_col == -1)) {
+		return; // Nothing under cursor.
+	}
 
-	TreeItem *selected = get_selected();
-	if (!selected)
-		return;
-	int ofs = get_item_offset(selected);
-	if (ofs == -1)
-		return;
+	const Size2 area_size = get_size() - cache.bg->get_minimum_size();
 
-	const int tbh = _get_title_button_height();
-	ofs -= tbh;
+	int y_offset = get_item_offset(selected_item);
+	if (y_offset != -1) {
+		const int tbh = _get_title_button_height();
+		y_offset -= tbh;
 
-	const int marginh = cache.bg->get_margin(MARGIN_TOP) + cache.bg->get_margin(MARGIN_BOTTOM);
-	int h = compute_item_height(selected) + cache.vseparation;
-	int screenh = get_size().height - h_scroll->get_combined_minimum_size().height - marginh - tbh;
+		const int cell_h = compute_item_height(selected_item) + cache.vseparation;
+		const int screen_h = area_size.height - h_scroll->get_combined_minimum_size().height - tbh;
 
-	if (h > screenh) { //screen size is too small, maybe it was not resized yet.
-		v_scroll->set_value(ofs);
-	} else if (ofs + h > v_scroll->get_value() + screenh) {
-		v_scroll->call_deferred("set_value", ofs - screenh + h);
-	} else if (ofs < v_scroll->get_value()) {
-		v_scroll->set_value(ofs);
+		if (cell_h > screen_h) { // Screen size is too small, maybe it was not resized yet.
+			v_scroll->set_value(y_offset);
+		} else if (y_offset + cell_h > v_scroll->get_value() + screen_h) {
+			v_scroll->call_deferred("set_value", y_offset - screen_h + cell_h);
```

Comment:
What's the reason for this to be deferred? It causes the edit box to be misaligned when editing an partially visible item.